src/time/src/lib.rs

```diff
@@ -99,7 +98,19 @@ pub extern "C" fn asctime_r(tm: *const tm, buf: *mut c_char) -> *mut c_char {
 
 #[no_mangle]
 pub extern "C" fn clock() -> clock_t {
-    unimplemented!();
+    let mut ts: timespec = unsafe { core::mem::uninitialized() };
+
+    if clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &mut ts) != 0 {
+        return -1;
+    }
+
+    if ts.tv_sec > time_t::max_value() / 1000000
```

Comment:
`1000000` would probably be easier to read as `1_000_000` like below.

Reply:
I actually originally wrote `1_000_000`; the autoformatter must have destroyed that without me noticing... (needs a config change?)
Once I get home I'll change it.